independent-projects/tools/devtools-common/src/main/java/io/quarkus/platform/tools/ToolsUtils.java

```diff
@@ -39,6 +40,44 @@ public static String getProperty(String name, String defaultValue) {
         return System.getProperty(name, defaultValue);
     }
 
+    public static boolean nonBlank(String arg) {
+        return !isBlank(arg);
+    }
+
+    public static boolean isBlank(String arg) {
+        return arg == null || arg.trim().isEmpty();
+    }
+
+    public static Map<String, String> stringToMap(
+            String str, String entrySeparator, String keyValueSeparator) {
+        try {
+            HashMap<String, String> result = new HashMap<>();
+            for (String entry : str.split(entrySeparator)) {
+                String[] pair = entry.split(keyValueSeparator);
+                if (pair.length > 2) {
+                    throw new IllegalArgumentException("Malformed entry " + entry);
+                }
+
+                if (pair.length == 1) {
+                    result.put(pair[0].trim(), "");
+                } else {
+                    if (isBlank(pair[0])) {
+                        throw new IllegalArgumentException("Entry with empty key " + entry);
+                    } else {
+                        result.put(pair[0].trim(), pair[1].trim());
+                    }
+                }
+            }
+            return result;
+        } catch (Exception ex) {
```

Comment:
I would prefer if you handle error cases than having a "random" try catch block.